packages/nuxt/src/core/app.ts

```diff
@@ -57,7 +57,9 @@ export async function generateApp (nuxt: Nuxt, app: NuxtApp) {
 export async function resolveApp (nuxt: Nuxt, app: NuxtApp) {
   // Resolve main (app.vue)
   if (!app.mainComponent) {
-    app.mainComponent = await findPath(['~/App', '~/app'])
+    app.mainComponent = await findPath(
+      nuxt.options._layers.flatMap(({ config: { srcDir } }) => [`${srcDir}/App`, `${srcDir}/app`])
```

Comment:
Code style: We avoid destructure context and config interfaces to make it easier tracing usages:

```suggestion
      nuxt.options._layers.flatMap((layer) => [`${layer.config.srcDir}/App`, `${layer.config.srcDir}/app`])
```